compiler/src/lexer.mll

```diff
@@ -135,10 +135,10 @@ rule main = parse
 
   (* Why this is needed *)
   | ((*'-'?*) digit+) as s   
-      { INT (Z.of_string s) } 
+      { INT {zvalue=Z.of_string s;raw=s}} 
 
   | ('0' ['x' 'X'] hexdigit+) as s
-      { INT (Z.of_string s) }
+      { INT {zvalue=Z.of_string s;raw=s} }
 
```

Comment:
Why do you compute the Z value here? Isn’t it simpler to only keep the string here and compute the value in pre-typing?

Reply:
I wasn't very used to the process of the compiler so I wanted the modification to affect the smallest possible part of the code. If you think it is preferable that the Z type building is done in pre-typing, I can change it (it will indeed simplify code readability)